states/remote/remote_test.go

```diff
@@ -126,3 +116,54 @@ func (c *mockClient) appendLog(method string, content []byte) {
 	}
 	c.log = append(c.log, mockClientRequest{method, contentVal})
 }
+
+// mockRemoteClient is like mockClient, but also implements
+// EnableForcePush, allowing testing for this behavior
+type mockRemoteClient struct {
+	current []byte
+	force   bool
+	log     []mockClientRequest
+}
```

Comment:
I was a bit confused by the name of this type, thinking "why isn't the `mockClient` also remote?" But is it "remote" as in "enhanced remote backend"?

Would naming the type something like `mockForceableClient` make sense to you? If my understanding of what's happening here is correct I think that would've been less confusing to me.

Reply:
OO yes I like that. I'll adopt that suggestion. I meant remote as in "like the remote backend" and agree that it's a bit too confusing as-is

Reply:
Adopted @mildwonkey's suggestion :)